pkg/component/component.go

```diff
@@ -8,22 +8,18 @@ import (
 
 // StoreAPI is a component that implements Thanos' gRPC StoreAPI.
 type StoreAPI interface {
-	implementsStoreAPI()
 	String() string
 	ToProto() storepb.StoreType
 }
 
 // Source is a Thanos component that produce blocks of metrics.
 type Source interface {
-	producesBlocks()
 	String() string
 }
 
 // SourceStoreAPI is a component that implements Thanos' gRPC StoreAPI
 // and produce blocks of metrics.
 type SourceStoreAPI interface {
-	implementsStoreAPI()
```

Comment:
I think it might be needed to ensure we have type safety (so someone cannot put StoreAPI as Source.

Having lowercase interface method should be fine, is it marked as wrong?

Reply:
I saw this trick recently used in Prometheus code - good observation. Will leave it there!